src/PhpSpec/CodeAnalysis/TokenizedTypeHintRewriter.php

```diff
@@ -192,4 +197,25 @@ private function tokenHasType($token, $type)
     {
         return is_array($token) && $type == $token[0];
     }
+
+    /**
+     * @param string $className
+     *
+     * @return bool
+     */
+    private function shouldExtractTokensOfClass($className)
+    {
+        return substr($className, -4) == 'Spec';
```

Comment:
The best option would be to have the `ResourceInterface` object inside this rewriter, but that's not possible because it come from a stream wrapper that just have the classname. 

The second option can be to use the `ResourceManager` from the classname. If a resource is found, then we extract the tokens from that class. What do you think about that?